### PR TITLE
Fixed license header files and enforced check at build time

### PR DESCRIPTION
**Pull request by JPinkney**
Fixed license header files and enforced check and build time.

Relevant issues: #329 and #192 

Signed-off-by: jpinkney <josh.pinkney@mail.utoronto.ca>

### CODE REVIEW

**Comment by l0rd** [comment]
Great @JPinkney! That was something that needed to be fixed since a long time. Can you please add copyright header to `*.sh` and `Dockerfile` as well?

File: builds/fabric8-che/pom.xml [hunk 1]
```diff
@@ -265,4 +265,22 @@
             <url>https://maven.codenvycorp.com/content/repositories/codenvy-public-snapshots/</url>
         </pluginRepository>
     </pluginRepositories>
+    <build>
```
Comment (sunix) [inline builds/fabric8-che/pom.xml]: Why do we need that ? isn't the license-maven-plugin inherited from upstream project ?

Also you should remove this part: https://github.com/redhat-developer/rh-che/blob/master/builds/fabric8-che/plugins/che-plugin-bayesian-lang-server/pom.xml#L84-L92

Comment (JPinkney) [reply builds/fabric8-che/pom.xml]: You were right, Its inherited from the upstream project. I also removed that part from the Bayesian language server pom.xml, I'm updating now

Comment (sunix) [reply builds/fabric8-che/pom.xml]: great :+1:

**Pull request merged**